Marlin/Configuration_adv.h

```diff
@@ -36,6 +36,23 @@
   #define THERMAL_PROTECTION_BED_HYSTERESIS 2 // Degrees Celsius
 #endif
 
+/**
+ * When a heater is switched off the temperature still rises for a little time, than cools down.
+ * When a heater is switched on the temperature may fall for a little while until the heat reaches the thermometer - than rises.
+ * MAX_TEMP_OVESTOOT_TIME is the longest time a system will need to reach the temperature again, it had when switched On/Off ( in seconds )
+ * This is used to check if a heater is really switched On/Off and the temperature reacts accordingly.
+ * OVERSTOOT is usually much longer than UNDERSHOOT - hence the name here.
+ * Set to 0 to deactivate this test
+ * Because when the heater is Off the temperature can not fall below the ambient temperature, what can differ, we need a limit.
+ * Hence we don't check for falling temperatures below MAX_AMBIENT_TEMP (in degree Celsius).
+ * Default is 50°C. If you have a heat-chamber you probably have to rise this.
+ * To not produce false positives we need some margin for noise at the ADCs
+ * MAX_SENSOR_NOISE_C is the maximum difference of temperature readings when real temperature does not change (in degree Celsius).
+ */
+#define MAX_TEMP_OVERSTOOT_TIME 180
```

Comment:
It isn't the time for it to warm up. It is the time for it to start moving
in the right direction, which I envisaged to be a few seconds. If that is
set to 3 minutes the check is useless because a hot end can easily get too
hot in 3 minutes so the algorithm becomes useless.

On 17 May 2015 at 07:00, ntoff <notifications@github.com> wrote:

> In Marlin/Configuration_adv.h
> <https://github.com/MarlinFirmware/Marlin/pull/2092#discussion_r30467718>:
>
> > @@ -36,6 +36,23 @@
> >    #define THERMAL_PROTECTION_BED_HYSTERESIS 2 // Degrees Celsius
> >  #endif
> >
> > +/**
> > + * When a heater is switched off the temperature still rises for a little time, than cools down.
> > + * When a heater is switched on the temperature may fall for a little while until the heat reaches the thermometer - than rises.
> > + * MAX_TEMP_OVESTOOT_TIME is the longest time a system will need to reach the temperature again, it had when switched On/Off ( in seconds )
> > + * This is used to check if a heater is really switched On/Off and the temperature reacts accordingly.
> > + * OVERSTOOT is usually much longer than UNDERSHOOT - hence the name here.
> > + * Set to 0 to deactivate this test
> > + * Because when the heater is Off the temperature can not fall below the ambient temperature, what can differ, we need a limit.
> > + * Hence we don't check for falling temperatures below MAX_AMBIENT_TEMP (in degree Celsius).
> > + * Default is 50°C. If you have a heat-chamber you probably have to rise this.
> > + * To not produce false positives we need some margin for noise at the ADCs
> > + * MAX_SENSOR_NOISE_C is the maximum difference of temperature readings when real temperature does not change (in degree Celsius).
> > + */
> > +#define MAX_TEMP_OVERSTOOT_TIME 180
>
> one of mine takes 5 or 6 minutes to get to 220, the other takes 2 and a
> half. Both hot ends similar so I guess if you have a rubbish heater
> cartridge then it can take a while to heat up. My slower one actually fails
> with the current dev version of Marlin.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/MarlinFirmware/Marlin/pull/2092/files#r30467718>.
>